hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/DFSPathSelector.java

```diff
@@ -98,35 +99,37 @@ public static DFSPathSelector createSourceSelector(TypedProperties props,
       long sourceLimit) {
 
     try {
-      // obtain all eligible files under root folder.
-      log.info("Root path => " + props.getString(Config.ROOT_INPUT_PATH_PROP) + " source limit => " + sourceLimit);
-      long lastCheckpointTime = lastCheckpointStr.map(Long::parseLong).orElse(Long.MIN_VALUE);
-      List<FileStatus> eligibleFiles = listEligibleFiles(fs, new Path(props.getString(Config.ROOT_INPUT_PATH_PROP)), lastCheckpointTime);
-      // sort them by modification time.
-      eligibleFiles.sort(Comparator.comparingLong(FileStatus::getModificationTime));
-      // Filter based on checkpoint & input size, if needed
-      long currentBytes = 0;
+      String pathStr = props.getString(Config.ROOT_INPUT_PATH_PROP);
       long maxModificationTime = Long.MIN_VALUE;
-      List<FileStatus> filteredFiles = new ArrayList<>();
-      for (FileStatus f : eligibleFiles) {
-        if (currentBytes + f.getLen() >= sourceLimit) {
-          // we have enough data, we are done
-          break;
+      if (!Boolean.valueOf(props.getString(Config.FULL_OVERWRITE,"false"))) {
+        // obtain all eligible files under root folder.
+        log.info("Root path => " + props.getString(Config.ROOT_INPUT_PATH_PROP) + " source limit => " + sourceLimit);
+        long lastCheckpointTime = lastCheckpointStr.map(Long::parseLong).orElse(Long.MIN_VALUE);
+        List<FileStatus> eligibleFiles = listEligibleFiles(fs, new Path(props.getString(Config.ROOT_INPUT_PATH_PROP)), lastCheckpointTime);
+        // sort them by modification time.
+        eligibleFiles.sort(Comparator.comparingLong(FileStatus::getModificationTime));
+        // Filter based on checkpoint & input size, if needed
+        long currentBytes = 0;
+        List<FileStatus> filteredFiles = new ArrayList<>();
+        for (FileStatus f : eligibleFiles) {
+          if (currentBytes + f.getLen() >= sourceLimit) {
+            // we have enough data, we are done
+            break;
+          }
+
+          maxModificationTime = f.getModificationTime();
+          currentBytes += f.getLen();
+          filteredFiles.add(f);
```

Comment:
IIUC, we can set 'sourceLimit' to `Long.MAX_VALUE` to read all of the data at one go. right?
if I am right we can save lots of changes